app/models/piece.rb

```diff
@@ -79,21 +65,83 @@ def moved?(x, y)
   end
 
   # Returns true if the coordinates provided
-  # have the same x-axis value.
-  def horizontal_move?(x)
-    x_distance(x) == 0
+  # have the same x-axis value and there are no
+  # pieces in between.
+  def clear_horizontal_move?(x, y)
+    return false unless y_distance(y) == 0
+    distance = x_distance(x)
+    path_clear?(x, y, distance)
   end
 
   # Returns true if the coordinates provided
-  # have the same y-axis value.
-  def vertical_move?(y)
-    y_distance(y) == 0
+  # have the same y-axis value and there are no
+  # pieces in between.
+  def clear_vertical_move?(x, y)
+    return false unless x_distance(x) == 0
+    distance = y_distance(y)
+    path_clear?(x, y, distance)
   end
 
   # Returns true if the coordinates provided
   # are the same distance away from the
-  # origin point along both axis.
-  def diagonal_move?(x, y)
-    x_distance(x) == y_distance(y)
+  # origin point along both axis and there are no
+  # pieces in between.
+  def clear_diagonal_move?(x, y)
+    return false unless x_distance(x) == y_distance(y)
+    distance = x_distance(x)
+    path_clear?(x, y, distance)
+  end
+
+  # Class constants used to determine path direction
+  # for obstruction logic.
+  RIGHT = 1
+  LEFT = -1
+  DOWN = 1
+  UP = -1
+
+  # Returns a hash with 1 or -1 values for
+  # x and y based on whether those values are
+  # increasing or decreasing towards destination
+  # to determine direction of path along both axis.
+  def path_direction(x, y)
+    direction = {}
+    direction[:x] = if x_position < x then RIGHT
+                    elsif x_position > x then LEFT
+                    else 0
+                    end
+    direction[:y] = if y_position < y then DOWN
+                    elsif y_position > y then UP
+                    else 0
+                    end
+    direction
+  end
+
+  # Returns an array of x-y coordinate subarrays
+  # between origin and destination.
+  def generate_path_coordinates(x, y, distance)
+    coordinate_sets = []
+    direction = path_direction(x, y)
+    (distance - 1).times do |i|
+      i += 1
+      coordinate_sets << [
+        x_position + i * direction[:x],
+        y_position + i * direction[:y]
```

Comment:
I don't think this is correct.  The default order of operations here is what we want -- `direction`'s sole job is to flip `i` between being a positive and a negative.  If you place parenthesis on the first two elements it produces a completely different result.  What I'm using `direction` for is to determine whether to increment up or down on an axis.

The example above is actually bad data because if `x_position` is 6, 'i' says we're on our 6th iteration which means we're counting at least 6 tiles between origin and destination, and 'direction' is -1 which means we're going left, then we are going off the board.

Let's take the example of a queen at [6, 0] making a move to [6, 7].  The call to the function would be `generate_path_coordinates(6, 7, 7)`.  This represents the largest distance you could move across the board.

`[73] pry(main)> queen
=> #<Queen:0xbb8a2378
 id: 1059,
 color: "white",
 type: "Queen",
 x_position: 6,
 y_position: 0,
 game_id: 35,
 player_id: nil,
 captured: nil,
 checkmate: nil,
 created_at: Fri, 29 Jul 2016 13:41:38 UTC +00:00,
 updated_at: Fri, 29 Jul 2016 13:41:38 UTC +00:00,
 moved: true>
[74] pry(main)> queen.generate_path_coordinates(6,7,7)
=> [[6, 1], [6, 2], [6, 3], [6, 4], [6, 5], [6, 6]]
[75] pry(main)>
`

Here is the incorrect return values with the parens added:

`[7] pry(main)> queen
=> #<Queen:0xbaddb688
 id: 1092,
 color: "white",
 type: "Queen",
 x_position: 6,
 y_position: 0,
 game_id: 36,
 player_id: nil,
 captured: nil,
 checkmate: nil,
 created_at: Fri, 29 Jul 2016 14:03:51 UTC +00:00,
 updated_at: Fri, 29 Jul 2016 14:03:51 UTC +00:00,
 moved: true>
[8] pry(main)> queen.generate_path_coordinates(6,7,7)
=> [[0, 1], [0, 2], [0, 3], [0, 4], [0, 5], [0, 6]]
[9] pry(main)>
`

As for tests, there is test coverage for all of the helper methods but it's all wound up in the final result.  I read somewhere that testing specific helper methods was bad practice and that I should just test the final methods in a way that it would bubble up bugs with helper methods.  As an aside, all tests correctly break when changing parens. :)

Let me know if this makes sense or if I'm in outer space.